src/mbgl/tile/vector_tile_data.cpp

```diff
@@ -77,6 +79,11 @@ VectorTileData::VectorTileData(const TileID& id_,
                 // existing buckets in case we got a refresh parse.
                 buckets = std::move(resultBuckets.buckets);
 
+                if (state == State::parsed) {
```

Comment:
In what situations does `parseGeometryTile` give you `TileParseResult` whose state isn't `parsed`?